### PR TITLE
Devas 1882 edit comment

### PR DESCRIPTION
**Pull request by pascalhuynh**
* add edit feature for a fiction comment (integration, unit tests, storybook and logic with the backend)

### CODE REVIEW

File: assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx [hunk 1]
```diff
@@ -90,20 +121,55 @@ export class FictionComment extends Component<LocalFictionCommentProps, FictionC
...
+  updateCommentHandler = (comment: string, commentId: string) => {
+    displayAlert('success', I18n.t('loading.wait'));
+    this.displayFictionCommentForm(false);
+    this.toggleIsEditing(false);
+
+    // Define variables
+    const { updateComment } = this.props;
+    const updatePostInputs: UpdateCommentInputs = {
+      body: comment,
+      contentLocale: 'fr',
```
Comment (amensouissi) [inline assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx]: fr ? use the global contentLocale

File: assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx [hunk 2]
```diff
@@ -182,15 +287,19 @@ const mapQueryToProps = ({ data }) => {
           ? fiction.creator.image.externalUrl
           : EMPTY_STRING
     };
+    const USER_ID_NOT_FOUND = -9999;
```
Comment (amensouissi) [inline assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx]: in the consts file?

File: assembl/static2/js/app/components/debate/brightMirror/fictionCommentForm.jsx [hunk 3]
```diff
@@ -28,12 +32,14 @@ const COMMENT_SUBMIT_BUTTON_ID = 'comment-submit-button';
...
   static defaultProps = {
+    commentValue: '',
+    editMode: false,
     rowsForTextarea: 2
   };
 
   state = {
-    commentTextareaValue: '',
-    showFormActionButtons: false
+    commentTextareaValue: this.props.commentValue ? this.props.commentValue : EMPTY_STRING,
```
Comment (amensouissi) [inline assembl/static2/js/app/components/debate/brightMirror/fictionCommentForm.jsx]: destructuring props

Comment (pascalhuynh) [reply assembl/static2/js/app/components/debate/brightMirror/fictionCommentForm.jsx]: @amensouissi that would force me to use 'constructor' method in order to destructure the props. Do you have an other syntax in mind ?

File: assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx [hunk 4]
```diff
@@ -90,20 +126,59 @@ export class FictionComment extends Component<LocalFictionCommentProps, FictionC
...
+    const updatePostInputs: UpdateCommentInputs = {
+      body: comment,
+      contentLocale: contentLocale,
+      postId: commentId
+    };
+
+    // Call the mutation function to update a comment
+    updateComment({ variables: updatePostInputs })
+      .then((result) => {
+        // If needed post result can be fetch with `result.data.createPost.post`
```
Comment (MilanBarande) [inline assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx]: fetched* :)

File: assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx [hunk 5]
```diff
@@ -123,43 +198,83 @@ export class FictionComment extends Component<LocalFictionCommentProps, FictionC
...
+    // Display ToggleCommentButton only when there are answers to a comment
     const displayToggleCommentButton = numChildren > 0 ? <ToggleCommentButton {...toggleCommentButtonProps} /> : null;
 
     // Display FictionCommentForm when ReplyToCommentButton is clicked.
     // ReplyToCommentButton is hidden when FictionCommentForm is displayed
-    const displayFictionCommentForm = showFictionCommentForm ? (
-      <FictionCommentForm {...fictionCommentFormProps} />
+    const displayReplyToCommentButton = showFictionCommentForm ? null : <ReplyToCommentButton {...replyToCommentButtonProps} />;
+    const displayFictionCommentForm = showFictionCommentForm ? <FictionCommentForm {...fictionCommentFormProps} /> : null;
+
```
Comment (MilanBarande) [inline assembl/static2/js/app/components/debate/brightMirror/fictionComment.jsx]: const answerForm (or a better name) = showFicitonCommentForm ? <FictionCommentForm {...fictionCommentFormProps} /> : <ReplyToCommentButton {...replyToCommentButtonProps} />;

**Pull request merged**